app/src/main/java/com/cmput301f20t21/bookfriends/ui/library/add/AddEditViewModel.java

```diff
@@ -132,13 +146,34 @@ public void handleEditBook(
         final String title = bookTitle.getValue();
         final String author = bookAuthor.getValue();
         final String description = bookDescription.getValue();
-        final Uri newUriFile = localImageUri.getValue();
-        final Boolean shouldDeleteImage = !hasImage;
+        final Uri imageUri = localImageUri.getValue();
 
-        bookRepository.editBook(oldBook, isbn, title, author, description, newUriFile, shouldDeleteImage).addOnSuccessListener(
+        // 3 image cases:
+        // 1. Update local image with a new image
+        // 2. Update local image with no image (book previously have image, but we removed it)
+        // 3. Did not touch image, keep old image
+        if (hasNewImage) {
+            if (imageUri != null) {     // if we have new image, we upload it to database then edit book
+                bookRepository.addImage(currentUsername, imageUri).addOnSuccessListener(imageUrl -> {
+                    bookRepository.editBook(oldBook, isbn, title, author, description, imageUrl).addOnSuccessListener(
+                            successCallback::run
+                    ).addOnFailureListener(e -> {
+                        failCallback.run(BOOK_ERROR.FAIL_TO_EDIT_BOOK);
+                    });
+                });
+            } else {    // if we removed the book image, we update the book with no image url
+                bookRepository.editBook(oldBook, isbn, title, author, description, null).addOnSuccessListener(
+                        successCallback::run
+                ).addOnFailureListener(e -> {
+                    failCallback.run(BOOK_ERROR.FAIL_TO_EDIT_BOOK);
+                });
```

Comment:
does this case delete the image from cloud storage?

Reply:
we could, but since we didn't previously, i didn't implement it.

Reply:
yeah this might stack up to our storage in the long run, but since this app is not going to be used by anyone we can leave it as it is😂

Reply:
we had the static image key previously so there was forever one image per book, users changing them or not. now that's a bigger issue but still we are 1000% fine to ignore it.